src/GraphQL/Utilities/Federation/FederatedSchemaBuilder.cs

```diff
@@ -119,7 +119,7 @@ private void AddTypeNameToSelection(Field field, Document document)
 
         private bool FindSelectionToAmend(SelectionSet selectionSet, Document document, out SelectionSet setToAlter)
         {
-            foreach (var selection in selectionSet.Selections)
+            foreach (var selection in selectionSet.SelectionsList)
```

Comment:
An example of an unnoticed place where it was possible to get rid of the iterator allocation for a long time.